src/ILVerify/src/ILImporter.Verify.cs

```diff
@@ -1267,7 +1267,7 @@ void ImportLoadIndirect(TypeDesc type)
             var address = Pop();
 
             if (type == null)
-                type = GetWellKnownType(WellKnownType.Object);
+                type = address.Type;
```

Comment:
The CheckIsByRef check may need to be moved up to avoid operating address.Type with assumption that it is managed pointer when it is actually not. Something like:

```
    CheckIsByRef(address);

    if (type == null)
    {
        type = address.Type;
        ...
    }
    else
    {
        CheckIsAssignable(address.Type, type);
    }
```